mongodump/barrier.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) MongoDB, Inc. 2025-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+package mongodump
+
+import (
+	"os"
+	"time"
+
+	"github.com/mongodb/mongo-tools/common/log"
+	"github.com/pkg/errors"
+)
+
+// Wait until a file exists and can be opened for reading
+// This is used only for testing mongodump/mongorestore with resmoke
+// test infrastructure.  The tests will create the barrier file when they have
+// finshed writes to the source cluster.
+func waitForSourceWritesDoneBarrier(barrierName string) {
+	log.Logvf(
+		log.DebugHigh,
+		"waitForSourceWritesDoneBarrier: initial check for existence of file %s",
+		barrierName,
+	)
+	start := time.Now()
+	logInterval := time.Minute
+	prevLogTime := start
+	for {
+		f, err := os.Open(barrierName)
```

Comment:
Nit: Do we care, really, about being able to _read_ the file, versus its mere existence?

If all we need to verify is its existence, without any need to _interact_ with the file, then a Stat() seems more apropos.

Reply:
As discussed here, the code gets more complicated for no real benefit if you go that way and try to get it right.

https://stackoverflow.com/questions/12518876/how-to-check-if-a-file-exists-in-go

Reply:
I would say it’s marginally simpler because you don’t have a file descriptor to close. You still have a command that can fail, and the error you check for is the same.

All you care about is existence. You might as well use the kernel API that gives you that information directly, IMO. It’s a small point, though, so I’ll close this thread.